mediapipe/graphs/instantmotiontracking/subgraphs/calculators/tracked_anchor_manager_calculator.cc

```diff
@@ -0,0 +1,154 @@
+      cc->Outputs().Tag(kAnchorsTag).Set<std::vector<Anchor>>();
+    }
+    if(cc->Outputs().HasTag(kCancelTag)) {
+      cc->Outputs().Tag(kCancelTag).Set<int>();
+    }
+
+    return ::mediapipe::OkStatus();
+  }
+
+  ::mediapipe::Status Open(CalculatorContext* cc) override {
+    cc->SetOffset(TimestampDiff(0));
+    return ::mediapipe::OkStatus();
+  }
+
+  ::mediapipe::Status Process(CalculatorContext* cc) override;
+};
+REGISTER_CALCULATOR(TrackedAnchorManagerCalculator);
+
+::mediapipe::Status TrackedAnchorManagerCalculator::Process(CalculatorContext* cc) {
+  std::vector<Anchor> current_anchor_data = cc->Inputs().Tag(kAnchorsTag).Get<std::vector<Anchor>>();
+
+  auto pos_boxes = absl::make_unique<TimedBoxProtoList>();
+
+  std::vector<Anchor> tracked_scaled_anchor_data;
+
+  // Update from tracking system or add anchor to tracking system
+  for(const Anchor previous_anchor : previous_anchor_data_) {
+    for(Anchor current_anchor : current_anchor_data) {
+      if(previous_anchor.sticker_id == current_anchor.sticker_id) {
+        // Check if anchor was repositioned by user, and reset tracking box
+        if(previous_anchor.x != current_anchor.x || previous_anchor.y != current_anchor.y) {
+          TimedBoxProto* box = pos_boxes->add_box();
+          box->set_left(current_anchor.x - box_h_w_/2);
+          box->set_right(current_anchor.x + box_h_w_/2);
+          box->set_top(current_anchor.y - box_h_w_/2);
+          box->set_bottom(current_anchor.y + box_h_w_/2);
+          box->set_id(current_anchor.sticker_id);
+          box->set_time_msec(cc->InputTimestamp().Microseconds() / 1000);
+          current_anchor.z = 1.0; // Default value for normalized z (scale factor)
+        }
+        // If anchor was not repositioned, update the location from the tracking system
+        else if (cc->Inputs().HasTag(kBoxesInputTag) && !cc->Inputs().Tag(kBoxesInputTag).IsEmpty()) {
+          for(const auto& box : cc->Inputs().Tag(kBoxesInputTag).Get<TimedBoxProtoList>().box())
+          {
+            if(box.id() == current_anchor.sticker_id) {
+              current_anchor.x = (box.left() + box.right())/2;
+              current_anchor.y = (box.top() + box.bottom())/2;
+              current_anchor.z = box_h_w_/(box.right() - box.left());
+            }
+          }
+        }
+        tracked_scaled_anchor_data.emplace_back(current_anchor);
+        break;
+      }
+    }
+  }
+
+  mediapipe::Timestamp timestamp = cc->InputTimestamp();
+  // Delete all boxes that are artifacts from deleted anchors
+  for(const TimedBoxProto& box : cc->Inputs().Tag(kBoxesInputTag).Get<TimedBoxProtoList>().box()) {
+    bool anchor_is_being_tracked = false;
+    for(Anchor point : tracked_scaled_anchor_data) {
+      if(box.id() == point.sticker_id) {
+        anchor_is_being_tracked = true;
+        break;
+      }
+    }
+    if(!anchor_is_being_tracked) {
+      cc->Outputs().Tag(kCancelTag).AddPacket(MakePacket<int>(box.id()).At(timestamp++));
```

Comment:
This is a neat trick for ensuring timestamp monotonicity across multiple simultaneously-released packets; since the behavior is a little weird, we should add a comment up at the top documenting how the CANCEL_ID stream works.

Technically, this is actually breaking the terms of using cc->SetOffset(TimestampDiff(0)) above, since we're manually manipulating the timestamp to have a diff other than 0 from the input, so we should probably also add a TODO to fix this behavior (one possible way to do so in the future would be to output a vector of box_id's).